include/yui/types.h

```diff
@@ -0,0 +1,142 @@
+#ifndef TYPES_H
+#define TYPES_H
+
+#include <imgui.h>
+
+#include <compare>
+
+namespace yui {
+
+/**
+ * @brief   Internal vec2 type
```

Comment:
This is in a public header, i.e not an internal type.

Reply:
Will be removed.